src/primitives/block.h

```diff
@@ -193,6 +195,26 @@ class CBlock : public CBlockHeader
     uint64_t GetBlockSize() const;
 };
 
+// Used for thin type blocks that we want to reconstruct
+class CBlock_ThinRelay : public CBlock
```

Comment:
I get why you used the underscore, but I think that we should stick with the standard bitcoin class naming convention, so in this case CBlockThinRelay.  The intention of these naming conventions is that capitalization and use of _ is hard to remember compared to the name itself.  With inconsistent naming, it becomes much harder to look things up from memory since most search tools aren't clever enough to find near matches.